src/hashstore/hashstore.py

```diff
@@ -114,11 +108,12 @@ def find_object(self, pid):
 
     @abstractmethod
     def store_metadata(self, pid, metadata, format_id):
-        """Add or update metadata, such as `sysmeta`, to disk using the given path/stream.
-
-        The `store_metadata` method uses a persistent identifier `pid` and a metadata `format_id`
-        to determine the permanent address of the metadata object. The permanent address is
-        calculated by obtaining the SHA-256 hex digest of the concatenation of `pid` & `format_id`.
+        """Add or update metadata, such as `sysmeta`, to disk using the given path/stream. The
+        `store_metadata` method uses a persistent identifier `pid` and a metadata `format_id`
+        to determine the permanent address of the metadata object. All metadata documents for a
+        given `pid` will be stored in a directory (under ../metadata) that is determined by
+        calculating the hash of the given pid, with the document name being the hash of the
+        metadata format (`format_id`).
```

Comment:
We are actually storing the metadata document with the name being the hash of the `pid` + `format_id`. This is so that the metadata document names in HashStore within a folder remain unique to reduce confusion.